inlang/source-code/sdk/src/storage/helper.ts

```diff
@@ -0,0 +1,52 @@
+import { Message, Variant } from "../versionedInterfaces.js"
+
+const fileExtension = ".json"
+
+export function getMessageIdFromPath(path: string) {
+	if (!path.endsWith(fileExtension)) {
+		return
+	}
+
+	const cleanedPath = path.replace(/\/$/, "") // This regex matches a trailing slash and replaces it with an empty string
+	const messageFileName = cleanedPath.split("/").join("_") // we split by the first leading namespace or _ separator - make sure slashes don't exit in the id
+	// const messageFileName = pathParts.at(-1)!
+
+	const lastDotIndex = messageFileName.lastIndexOf(".")
+
+	// Extract until the last dot (excluding the dot)
+	return messageFileName.slice(0, Math.max(0, lastDotIndex))
+}
+
+export function getPathFromMessageId(id: string) {
+	const path = id.replace("_", "/") + fileExtension
+	return path
+}
+
```

Comment:
Exporting individual functions instead of nesting them in objects like `utils.getMessageIdFromPath` enables tree-shaking. Good choice 👍